models/UserManager.php

```diff
@@ -0,0 +1,289 @@
+        $chars = "azertyuiopqsdfghjklmwxcvbnAZERTYUIOPQSDFGHJKLMWXCVBN0123456789";
+        
+        return substr(str_shuffle(str_repeat($chars,$length)),0,$length);
+    }
+
+    // Fonction permettant d'ajouter un modérateur / administrateur et de lui envoyer un mail
+    function add_modo($name,$email,$role,$token){
+        $tableau= [
+            'name'      =>  $name,
+            'email'     =>  $email,
+            'token'     =>  $token,
+            'role'      =>  $role
+        ];
+
+        $sql = "INSERT INTO admins(name,email,token,role) VALUES(:name,:email,:token,:role)";
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+
+        $subject = "Modo / Admin sur le blog";
+        $message = '              
+           <html lang="en" style="font-family: sans-serif;">
+                <head>
+                    <meta charset="UTF-8">
+                </head>
+                <body>
+                    Voici votre identifiant et code unique à insérer sur <a href="http://projet-5.jeromelacquemant.fr/index.php?page=new">cette page</a>.
+                    <br/>Identifiant: '.$email.'
+                    <br/>Mot de passe: '.$token.'
+                    <br/>Vous êtes: '.$role.'
+                    <br/><br/>Après avoir inséré ces informations, vous devrez choisir un mot de passe.
+                </body>
+            </html>
+        ';
+
+        $header = "MIME-Version: 1.0\r\n";
+        $header .= "Content-type: text/html; charset=UTF-8\r\n";
+        $header .= 'From: jerome.lacquemant@gmail.com' . "\r\n" . 'Reply-To: jerome.lacquemant@gmail.com' . "\r\n" . 'X-Mailer: PHP/' . phpversion();
+
+        mail($email,$subject,$message,$header);
+}
+    
+    // Fonction permettant d'obtenir tous les modérateurs / administrateurs
+    function get_modos(){
+         $req = $this->db->query("
+            SELECT * FROM admins
+        ");
+
+        $results = [];
+        while($rows = $req->fetchObject()){
+            $results[] = $rows;
+        }
+        
+        return $results;
+    }
+
+    // Fonction qui vérifie qu'un utilisateur est bien modérateur ou admin
+    function is_modo($email,$token){
+        $tableau = [
+            'email' =>  $email,
+            'token' =>  $token
+        ];
+        $sql = "SELECT * FROM admins WHERE email=:email AND token=:token";
+        $req= $this->db->prepare($sql);
+        $req->execute($tableau);
+        
+        return $req->rowCount($sql);
+    }
+
+    // Fonction permettant la mise à jour du mot de passe 
+    function update_password($password){
+         $tableau = [
+            'password'  =>  sha1($password),
+            'session'   =>  $_SESSION['admin']
+        ];
+
+        $sql = "UPDATE admins SET password = :password WHERE email=:session";
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+
+    }
+
+    // Fonction permettant de récupérer la liste des administrateurs
+    function get_user(){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [get_user accesses the super-global variable $_SESSION.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)